test/cluster/test_alternator_proxy_protocol.py

```diff
@@ -0,0 +1,318 @@
+# Copyright 2025-present ScyllaDB
+#
+# SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+
+"""
+Tests for Alternator (DynamoDB API) ports with proxy protocol v2 support.
+
+These tests verify that when Scylla is configured with Alternator proxy protocol
+ports, clients can connect using the PROXY protocol v2 header.
+"""
+
+import datetime
+import hashlib
+import hmac
+import json
+import logging
+import socket
+import ssl
+import struct
+from typing import Optional
+
+import aiohttp
+import pytest
+
+from test.pylib.manager_client import ManagerClient
+
+logger = logging.getLogger(__name__)
+
+# Proxy protocol v2 signature (12 bytes)
+PROXY_V2_SIGNATURE = b'\x0d\x0a\x0d\x0a\x00\x0d\x0a\x51\x55\x49\x54\x0a'
+
+# Port numbers
+ALTERNATOR_PORT = 8000
+ALTERNATOR_HTTPS_PORT = 8043
+ALTERNATOR_PROXY_PORT = 28000
+ALTERNATOR_HTTPS_PROXY_PORT = 28043
+
+
+def make_proxy_v2_header(src_addr: str, src_port: int, dst_addr: str, dst_port: int) -> bytes:
+    """
+    Construct a proxy protocol v2 header for IPv4 TCP connections.
+    See: https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt
+    """
+    header = bytearray()
+    header.extend(PROXY_V2_SIGNATURE)
+    header.append(0x21)  # Version 2, PROXY command
+    header.append(0x11)  # AF_INET, STREAM/TCP
+    header.extend(struct.pack('!H', 12))  # Address block length
+    header.extend(socket.inet_aton(src_addr))
+    header.extend(socket.inet_aton(dst_addr))
+    header.extend(struct.pack('!H', src_port))
+    header.extend(struct.pack('!H', dst_port))
+    return bytes(header)
+
+
+def sign_request(method: str, host: str, uri: str, headers: dict, body: str,
+                 access_key: str, secret_key: str,
+                 region: str = 'us-east-1', service: str = 'dynamodb') -> dict:
+    """Sign an AWS request using Signature Version 4."""
```

Comment:
Nice that you also checked signed requests, although I don't see why it should have made any difference with the proxy protocol, since if I remember correctly, the AWS signature does not use IP addresses or ports at all, so the fact the proxy protocol modify them doesn't make a difference.

By the way, in test/alternator/util.py we already have a function get_signed_request() that does the AWS signature v4 protocol - but it uses boto3's weird "request signer" abstraction. Your implementation is duplicates this feature, but I like it - it's cleaner, more self contained, and more reusable because it just uses sha256 and hmac, and doesn't look like mysterious magic incantations. I like it.